allel/io/vcf_read.py

```diff
@@ -1008,46 +1009,25 @@ def vcf_to_zarr(input, output,
 
 
 # noinspection PyShadowingBuiltins
-def _setup_input_stream(input, region=None, tabix=None, buffer_size=DEFAULT_BUFFER_SIZE):
+def _setup_input_stream(input, index=None, region=None, buffer_size=DEFAULT_BUFFER_SIZE):
 
     # obtain a file-like object
     close = False
     if isinstance(input, str) and input.endswith('gz'):
+        if index is None:
+            index_file = '{}.tbi'.format(input)
+            if os.path.exists(index_file):
+                index = index_file
 
-        if region and tabix and os.name != 'nt':
-
-            try:
-                # try tabix
-                p = subprocess.Popen([tabix, '-h', input, region],
-                                     stdout=subprocess.PIPE,
-                                     stderr=subprocess.STDOUT,
-                                     bufsize=0)
-
-                # check if tabix exited early, look for tabix error
-                time.sleep(.5)
-                poll = p.poll()
-                if poll is not None and poll > 0:
-                    err = p.stdout.read()
-                    err = str(err, 'ascii')
-                    p.stdout.close()
-                    raise RuntimeError(err.strip())
-                fileobj = p.stdout
-                close = True
-                # N.B., still pass the region parameter through so we get strictly only
-                # variants that start within the requested region. See also
-                # https://github.com/alimanfoo/vcfnp/issues/54
-
-            except FileNotFoundError:
-                # no tabix, fall back to scanning
-                warnings.warn('tabix not found, falling back to scanning to region')
-                fileobj = gzip.open(input, mode='rb')
-                close = True
-
-            except Exception as e:
-                warnings.warn('error occurred attempting tabix (%s); falling back to '
-                              'scanning to region' % e)
-                fileobj = gzip.open(input, mode='rb')
-                close = True
+        if index is not None and os.path.getmtime(index) < os.path.getmtime(input):
```

Comment:
Small point, but maybe this check could be moved so that it's only performed if user supplies a `region` argument. I.e., only check the mtime of the index file if it's actually going to be used.

Reply:
you make the point, this check should move into to fellow if block, my fault